java/src/jmri/jmrit/display/layoutEditor/LayoutEditorDialogs/LayoutTurntableEditor.java

```diff
@@ -217,20 +297,102 @@ private void editLayoutTurntableEditBlockPressed(ActionEvent a) {
 
     // Remove old rays and add them back in
     private void updateRayPanel() {
-        for (Component comp : editLayoutTurntableRayPanel.getComponents()) {
-            editLayoutTurntableRayPanel.remove(comp);
-        }
-
         // Create list of turnouts to be retained in the NamedBeanComboBox
         turntableTurnouts.clear();
-        for (LayoutTurntable.RayTrack rt : layoutTurntable.getRayTrackList()) {
-            turntableTurnouts.add(rt.getTurnout());
-        }
+        layoutTurntable.getRayTrackList().forEach(rt -> turntableTurnouts.add(rt.getTurnout()));
 
+        editLayoutTurntableRayPanel.removeAll();
         editLayoutTurntableRayPanel.setLayout(new BoxLayout(editLayoutTurntableRayPanel, BoxLayout.Y_AXIS));
         for (LayoutTurntable.RayTrack rt : layoutTurntable.getRayTrackList()) {
             editLayoutTurntableRayPanel.add(new TurntableRayPanel(rt));
         }
+
+        // Rebuild signal mast panel
+        signalMastParametersPanel.removeAll();
+        approachMastComboBoxes.clear();
+
+        if (layoutTurntable.isDispatcherManaged()) {
+            signalMastParametersPanel.setLayout(new BoxLayout(signalMastParametersPanel, BoxLayout.Y_AXIS));
+
+            // Add approach masts for each ray
+            for (LayoutTurntable.RayTrack rt : layoutTurntable.getRayTrackList()) {
+                JPanel p = new JPanel(new FlowLayout(FlowLayout.LEFT));
+                p.add(new JLabel(Bundle.getMessage("ApproachMastRay", rt.getConnectionIndex() + 1)));
+                NamedBeanComboBox<SignalMast> combo = new NamedBeanComboBox<>(
+                        InstanceManager.getDefault(SignalMastManager.class), rt.getApproachMast(), DisplayOptions.DISPLAYNAME);
+                LayoutEditor.setupComboBox(combo, false, true, true);
+                combo.setEditable(false);
+                combo.setAllowNull(true);
+                p.add(combo);
+                signalMastParametersPanel.add(p);
+                approachMastComboBoxes.add(combo);
+            }
+
+            // Add action listeners now that the list is complete
+            for (int i = 0; i < approachMastComboBoxes.size(); i++) {
+                final int index = i; // final variable for use in lambda
+                approachMastComboBoxes.get(i).addActionListener(e -> {
```

Comment:
The following line needs to be added before the add listener.

`approachMastComboBoxes.get(i).setExcludedItems(mastsUsedElsewhere);`